scripts/portal-fetcher/openshift-docs-downloader.py

```diff
@@ -0,0 +1,1186 @@
+                simple_path = f"{simple_path}/index.html"
+            elif not path.endswith('.html'):
+                simple_path = f"{simple_path}/index.html"
+            
+            local_path = output_dir / simple_path
+        else:
+            # Fallback if structure is unexpected
+            if path.endswith('/'):
+                path = path + "index.html"
+            elif not path.endswith('.html'):
+                path = path + "/index.html"
+            local_path = output_dir / path.lstrip("/")
+    except (ValueError, IndexError):
+        # Fallback if html-single not in path
+        if path.endswith('/'):
+            path = path + "index.html"
+        elif not path.endswith('.html'):
+            path = path + "/index.html"
+        local_path = output_dir / path.lstrip("/")
+    
+    # Ensure parent directory exists
+    local_path.parent.mkdir(parents=True, exist_ok=True)
+    
+    return local_path
+
+# Database functions
+
+def init_database(db_path):
+    """Initialize SQLite database to track downloaded files"""
+    Path(db_path).parent.mkdir(parents=True, exist_ok=True)
+    
+    conn = sqlite3.connect(db_path)
+    cursor = conn.cursor()
+    
+    # Main downloads table
+    cursor.execute('''
+    CREATE TABLE IF NOT EXISTS downloads (
+        id INTEGER PRIMARY KEY AUTOINCREMENT,
+        url TEXT UNIQUE,
+        local_path TEXT,
+        status TEXT,
+        etag TEXT,
+        last_modified TEXT,
+        timestamp DATETIME DEFAULT CURRENT_TIMESTAMP
+    )
+    ''')
+    
+    # History table to track changes
+    cursor.execute('''
+    CREATE TABLE IF NOT EXISTS download_history (
+        id INTEGER PRIMARY KEY AUTOINCREMENT,
+        url TEXT,
+        local_path TEXT,
+        status TEXT,
+        etag TEXT,
+        last_modified TEXT,
+        timestamp DATETIME DEFAULT CURRENT_TIMESTAMP,
+        change_type TEXT
+    )
+    ''')
+    
+    conn.commit()
+    conn.close()
+    logger.info(f"Database initialized at {db_path}")
+    
+    return db_path
+
+def record_download(db_path, url, local_path, status="success", etag=None, last_modified=None, change_type=None):
+    """
+    Record a download in the database and update history
+    
+    Args:
+        db_path (Path): Path to SQLite database
+        url (str): URL of the downloaded page
+        local_path (str/Path): Local path where the file is saved
+        status (str): Download status (success, failed, error)
+        etag (str): ETag header from the response, if available
+        last_modified (str): Last-Modified header from the response, if available
+        change_type (str): Type of change (new, updated, unchanged, error)
+    """
+    conn = sqlite3.connect(db_path)
+    cursor = conn.cursor()
+    local_path_str = str(local_path)
+    
+    try:
+        # Check if the record already exists
+        cursor.execute("SELECT * FROM downloads WHERE url = ?", (url,))
+        existing_record = cursor.fetchone()
```

Comment:
existing_record is not used. Not sure what's the purpose of checking if the record exists.